packages/flutter/test/cupertino/checkbox_test.dart

```diff
@@ -466,4 +468,102 @@ void main() {
         ),
     );
   });
+
+  testWidgets('Checkbox fill color resolves in enabled/disabled states', (WidgetTester tester) async {
+    const Color activeEnabledFillColor = Color(0xFF000001);
+    const Color activeDisabledFillColor = Color(0xFF000002);
+
+    Color getFillColor(Set<WidgetState> states) {
+      if (states.contains(WidgetState.disabled)) {
+        return activeDisabledFillColor;
+      }
+      return activeEnabledFillColor;
+    }
+
+    final WidgetStateProperty<Color> fillColor =
+      WidgetStateColor.resolveWith(getFillColor);
+
+    Widget buildApp({required bool enabled}) {
+      return CupertinoApp(
+        home: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) {
+              return CupertinoCheckbox(
+                value: true,
+                fillColor: fillColor,
+                onChanged: enabled ? (bool? value) { } : null,
+              );
+          },
+        ),
+      );
+    }
+
+    RenderBox getCheckboxRenderer() {
+      return tester.renderObject<RenderBox>(find.byType(CupertinoCheckbox));
+    }
+
+    await tester.pumpWidget(buildApp(enabled: true));
+    await tester.pumpAndSettle();
+    expect(getCheckboxRenderer(), paints..path(color: activeEnabledFillColor));
+
+    await tester.pumpWidget(buildApp(enabled: false));
+    await tester.pumpAndSettle();
+    expect(getCheckboxRenderer(), paints..path(color: activeDisabledFillColor));
+  });
+
+  testWidgets('Checkbox fill color resolves in hovered/focused states', (WidgetTester tester) async {
+    final FocusNode focusNode = FocusNode(debugLabel: 'checkbox');
+    addTearDown(focusNode.dispose);
+
+    tester.binding.focusManager.highlightStrategy = FocusHighlightStrategy.alwaysTraditional;
+    const Color hoveredFillColor = Color(0xFF000001);
+    const Color focusedFillColor = Color(0xFF000002);
+    const Color transparentColor = Color(0x00000000);
+
+    Color getFillColor(Set<WidgetState> states) {
+      if (states.contains(WidgetState.hovered)) {
+        return hoveredFillColor;
+      }
+      if (states.contains(WidgetState.focused)) {
+        return focusedFillColor;
+      }
+      return transparentColor;
+    }
+
+    final WidgetStateProperty<Color> fillColor =
+      WidgetStateColor.resolveWith(getFillColor);
```

Comment:
nit:

```suggestion
    final WidgetStateProperty<Color> fillColor = WidgetStateColor.resolveWith(getFillColor);
```

(the [style guide](https://github.com/flutter/flutter/blob/8c17a747c979818ce79dbaeaa64880e3d6d4b025/docs/contributing/Style-guide-for-Flutter-repo.md#prefer-a-maximum-line-length-of-80-characters) recommends no breaks after assignment operators)

Reply:
or, alternatively:

```suggestion
    final WidgetStateProperty<Color> fillColor =
        WidgetStateColor.resolveWith(getFillColor);
```

(the double-indent would make it consistent with Dart's auto-formatter & other framework code)